src/widgets/AutocompleteSelector.js

```diff
@@ -77,50 +69,48 @@ export class AutocompleteSelector extends React.PureComponent {
    * object (has the filtered member) or if it is an array. Otherwise,
    * return an empty list to display.
    */
-  getData = () => {
-    const { options, primaryFilterProperty, queryString } = this.props;
-    if (options && options.filtered && queryString) return this.filterResultData(options);
-    if (Array.isArray(options) && primaryFilterProperty) return this.filterArrayData(options);
+  const getData = () => {
+    if (options && options.filtered && queryString) return filterResultData(options);
+    if (Array.isArray(options) && primaryFilterProperty) return filterArrayData(options);
     return [];
   };
 
-  render() {
-    const { onSelect, placeholderText, renderLeftText, renderRightText } = this.props;
-
-    const data = this.getData();
-
-    return (
-      <View style={localStyles.container}>
-        <SearchBar
-          autoCapitalize="none"
-          autoCorrect={false}
-          autoFocus
-          color="white"
-          onChange={text => this.setState({ queryText: text })}
-          placeholder={placeholderText}
-          placeholderTextColor="white"
-          style={[localStyles.text, localStyles.searchBar]}
+  const data = getData();
+
+  const ResultRowWithOnePress = withOnePress(ResultRow);
+
+  return (
+    <View style={localStyles.container}>
+      <SearchBar
+        autoCapitalize="none"
+        autoCorrect={false}
+        autoFocus
+        color="white"
+        onChange={text => setQueryText(text)}
+        placeholder={placeholderText}
+        placeholderTextColor="white"
+        style={[localStyles.text, localStyles.searchBar]}
+      />
+      {data.length > 0 && (
+        <FlatList
+          data={data}
+          keyExtractor={item => item.id || item.name}
```

Comment:
could we define this explicitly so re-renders don't cause a new function to be created?

Reply:
Resolved with 81441a9